server/controllers/contest.js

```diff
@@ -63,20 +63,67 @@ exports.getContest = asyncHandler(async (req, res, next) => {
 // handler for getting all the contests
 exports.getContests = asyncHandler(async (req, res, next) => {
   const contestList = [];
+  const { search, title, startTime, endTime } = req.query;
   try {
-    const contests = await Contest.find({}).populate('userId');
-    contests.forEach((contest) => {
-      const contestData = {
-        _id: contest._id,
-        ownerName: contest.userId.username,
-        profileImg: contest.userId.profilePicUrl,
-        title: contest.title,
-        description: contest.description,
-        prizeAmount: contest.prizeAmount,
-      };
-      contestList.push(contestData);
-    });
-    res.status(200).json({ contest: contestList });
+    if (search === 'null' && title === 'null' && startTime === 'null' && endTime === 'null') {
+      const contests = await Contest.find({}).populate('userId');
+      contests.forEach((contest) => {
+        const contestData = {
+          _id: contest._id,
+          ownerName: contest.userId.username,
+          profileImg: contest.userId.profilePicUrl,
+          title: contest.title,
+          description: contest.description,
+          prizeAmount: contest.prizeAmount,
+        };
+        contestList.push(contestData);
+      });
+      res.status(200).json({ contest: contestList });
+    } else if (search !== 'null') {
```

Comment:
Wouldn't we also want to check if the startTime/endTime are null over here?

Reply:
I think that it doesn't need to check because there are difference handle submit function: handleSimpleSubmit and handleSubmit, in handleSimpleSubmit , only the search params won't be null, and in handleSubmit, search params must be null.